bots/notify/src/main/java/org/openjdk/skara/bots/notify/NotifyBot.java

```diff
@@ -83,6 +84,13 @@ public String toString() {
     public List<WorkItem> getPeriodicItems() {
         var ret = new ArrayList<WorkItem>();
 
+        if (!wildCardProtected && prListeners.stream()
+                .anyMatch(e -> e instanceof PullRequestBranchNotifier && ((PullRequestBranchNotifier) e).isProtectBranches())) {
+            log.info("Protecting branch *");
+            repository.protectBranchPattern("*");
+            wildCardProtected = true;
+        }
```

Comment:
I would suggest adding a new method on the `PullRequestListener` interface `initialize(HostedRepository)` and implement this method in `PullRequestBranchNotifier` so you can put this logic there instead. Then you can call `initialize` on all listeners when `getPeriodicItems` is called the first time.

Reply:
Good idea. Will do it.